os/tools/mksyscall.c

```diff
@@ -246,9 +248,10 @@ static void generate_proxy(int nparms)
 
 	fprintf(stream, "#include <syscall.h>\n\n");
 
-	cond_parm = get_parm(COND_INDEX);
-	if (cond_parm[0] != '\0')
+	if (strcmp(get_parm(NAME_INDEX),"ioctl")==0) {
```

Comment:
Please create a new api called `check_special_case()` and call it here. 
In this api, you can check for ioctl api. In future if we need to add more checks, we can add in the api. 
The api must return true if it is a special case.